content/blog/2020/05/2020-05-07-docker-windows-agents.adoc

```diff
@@ -0,0 +1,121 @@
+* `-DirectConnection <HOST:PORT>` or JENKINS_DIRECT_CONNECTION - Connect directly to this TCP agent port, skipping the HTTP(S) connection parameter download.
+* `-InstanceIdentity <BASE64VALUE>` or JENKINS_INSTANCE_IDENTITY - The base64 encoded InstanceIdentity byte array of the Jenkins master. When this is set, the agent skips connecting to an HTTP(S) port for connection info
+* `-JavaHome <JAVA_HOME>` or JAVA_HOME  - An override for the default JAVA_HOME baked into the image (different for JDK8 vs. JDK11)
+* `-Protocols <PROTOCOLS>` or JENKINS_PROTOCOLS - Specify the link:https://github.com/jenkinsci/remoting/blob/de7818885a5bf478760ba29f5ee216291437cb16/docs/protocols.md#active-protocols[remoting protocols] to attempt when instanceIdentity is provided.
+
+[NOTE]
+====
+There is currently an issue where the `-Protocols` parameter is not actually passed to agent.jar. The fix has been merged, but a release has not been made.
+====
+
+Example:
+
+```
+docker run jenkins/inbound-agent:windowsservercore-1809 -Url http://jenkins-server:port -WorkDir=C:/Users/jenkins/Agent -Secret <SECRET> -Name <AGENTNAME>
+```
+
+Example using environment variables:
+
+```
+docker run -e "JENKINS_URL=http://jenkins-server:port" -e "JENKINS_AGENT_NAME=AGENTNAME" jenkins/inbound-agent:windowsservercore-1809 -WorkDir=C:/Users/jenkins/Agent -Secret <SECRET> -Name <AGENTNAME>
+```
+
+The URL, Name and Secret parameters are required, but can be specified as either command line parameters or environment variables.
+
+
+
+=== jenkins/ssh-agent
+
+As mentioned above the jenkins/ssh-agent docker image is based on SSH communication with the master, rather than the remoting TCP or WebSocket protocols. This communication is initiated by the master rather than the agent connecting to the master is done with jenkins/agent and jenkins/inbound-agent.
+
+The image sets up a `jenkins` user and the OpenSSH server so that the master can connect to the agent via SSH. The image expects an SSH public key as a parameter and puts that key into the authorized_keys file for the jenkins user. The private key should be specified in the agent configuration on the master to allow the master to connect correctly.
+
+Example:
+
+```
+docker run jenkins/ssh-agent "<public key>"
+```
+
+You can also pass the public key as an environment variable when using `docker run`.
+
+Example:
+
+```
+docker run -e "JENKINS_AGENT_SSH_PUBKEY=<public key>" jenkins/ssh-agent
+```
+
+You will then be able to connect this agent using the plugin:ssh-slaves[SSH Build Agents Plugin] as "jenkins" with the matching private key.
+
+
+
+== What's next?
+
+There is an link:github.com/jenkinsci/docker/pull/924[open PR] to create a Windows based Docker image for a Jenkins master. There hasn't been a lot of requests for this, but to make the offerings complete for Windows users, the PR was created.
```

Comment:
```suggestion
There is an link:github.com/jenkinsci/docker/pull/924[open pull-request] to create a Windows based Docker image for a Jenkins master. There hasn't been a lot of requests for this, but to make the offerings complete for Windows users, the PR was created.
In the case of agents, we also consider providing versions based on _Windows 10, version 1909_ so that Jenkins users could run these images on GKE clusters
(see link:https://github.com/jenkinsci/docker-agent/issues/134[this issue]).

For plans unrelated to Windows, please see the link:/blog/2020/05/06/docker-agent-image-renaming/[Docker images for agents: New names and What's next] blogpost.
```

Follow-up from the platform SIG meeting discussion about GKE.